### PR TITLE
Add PyPI upload CI and release checklist

### PR DESCRIPTION
**Pull request by brynpickering**
Automates the release to PyPI, including a safety CI to release to Test-PyPI in a release PR before uploading to PyPI when a release is tagged and then published.

Summary of changes in this pull request:

* CI updates
* Release checklist that aligns with CI

Reviewer checklist:

- [ ] Test(s) added to cover contribution
- [ ] Documentation updated
- [ ] Changelog updated
- [ ] Coverage maintained or improved

### CODE REVIEW

**Comment by brynpickering** [comment]
I should point out that the CI _may_ not actually work properly, but it's difficult to test until it's merged in. So we may need to iterate on the CI jobs when we see how it reacts to our alpha release PR.

File: docs/contributing.md [hunk 1]
```diff
@@ -209,7 +209,23 @@ When adding docstrings, we request you use the [Google docstring style](https://
...
+- [ ] Commit your changes and create a release pull request with the title `Release vX.Y.Z` to have all the tests run and to verify that the pip package builds successfully.
+- [ ] Once the PR is approved and merged, tag the commit in main with the version `vX.Y.Z`.
+- [ ] Create a release through the GitHub web interface, using the same tag, titling it `Release vX.Y.Z` and include all the changelog elements that are `User-facing`.
+
+!!! note
+    The pull request _must_ have the title `Release vX.Y.Z` to trigger the pip package build and the test-pypi upload.
+
+### Post-release
+
+- [ ] Update the changelog, adding a new `Unreleased` heading.
```
Comment (sjpfenninger) [inline docs/contributing.md:226]:
```suggestion
- [ ] Update the `CHANGELOG.md`, adding a new `Unreleased` heading.
```

File: docs/contributing.md [hunk 2]
```diff
@@ -209,7 +209,23 @@ When adding docstrings, we request you use the [Google docstring style](https://
 
 ## Release checklist
 
-<!--- TODO -->
+### Create release
+
+- [ ] Create a release branch
+- [ ] Bump the version number in `src/calliope/_version.py`
+- [ ] Update the CHANGELOG.md with final version number of the form `vX.Y.Z` and the release date.
```
Comment (sjpfenninger) [inline docs/contributing.md:216]:
```suggestion
- [ ] Update the `CHANGELOG.md` with final version number of the form `vX.Y.Z` and the release date.
```

**Review by sjpfenninger — DISMISSED** [review]
It looks good to me - we'll have to see if it actually works exactly as described!

**Pull request merged**